aten/src/ATen/native/mps/operations/Blas.mm

```diff
@@ -51,13 +51,35 @@ Tensor dot_mps(
           MPSGraphTensor *selfTensor = mps::mpsGraphRankedPlaceHolder(mpsGraph, self);
           MPSGraphTensor *otherTensor =  mps::mpsGraphRankedPlaceHolder(mpsGraph, other);
 
-          MPSGraphTensor *dot = [mpsGraph multiplicationWithPrimaryTensor: selfTensor
-                                                          secondaryTensor: otherTensor
+          MPSGraphTensor *castSelf = nil;
+          MPSGraphTensor *castOther = nil;
+
+          if(self.scalar_type() == ScalarType::Short || self.scalar_type() == ScalarType::Byte) {
```

Comment:
Why only these types? What about `ScalarType::Char`, `ScalarType::Bool` or `ScalarType::Long`?

Reply:
Can you at least add the Char (not sure about Bool)?

Reply:
Dot doesn't support  bool, I added support for char